src/js/letters/components/Address.jsx

```diff
@@ -60,6 +60,37 @@ class Address extends React.Component {
     return lowerCity === 'apo' || lowerCity === 'fpo' || lowerCity === 'dpo';
   }
 
+  adjustStateNames(stateList, militaryStates, city) {
+    // Reformat the state name data so that it can be
+    // accepted by ErrorableSelect,
+    // e.g., from this: `IL: 'Illinois'`
+    // to this: `{ value: 'Illinois', label: 'IL' }`
+    let states = [];
+    _.mapKeys(stateList, (value, key) => {
+      states.push({ label: value, value: key });
+    });
+    // Add military states to full state list
+    militaryStates.forEach((militaryState) => {
+      states.push(militaryState);
+    });
+    // Alphabetize the list
+    states.sort((a, b) => {
+      if (a.label < b.label) {
+        return -1;
+      }
+      if (a.label > b.label) {
+        return 1;
+      }
+      return 0;
+    });
+    // Filter state list if military city is entered
+    if (this.props.value[city] && this.isMilitaryCity(this.props.value[city])) {
+      states = states.filter(state => state.value === 'AE' || state.value === 'AP' || state.value === 'AA');
```

Comment:
Should we save the filtering and just use the `militaryStateNames` from `helpers.jsx`? Also, if we do that, we should probably gate setting `states` based on whether a military city is selected:

```js
let states = [];
if (this.props.value[city] && this.isMilitaryCity(this.props.value[city])) {
  states = militaryStateNames;
} else {
  _.mapKeys(stateList, (value, key) => {
    states.push({ label: value, value: key });
  });
  // Add military states to full state list
  // Would this still be necessary?
  militaryStates.forEach((militaryState) => {
    states.push(militaryState);
  });
}
```
That way we don't sort all the states and then discard them, y'know? Avoids wasted cpu cycles.

Reply:
Ah gotcha, makes sense!